crates/build/src/lib.rs

```diff
@@ -370,6 +371,65 @@ fn exec_cargo_for_onchain_target(
     Ok(())
 }
 
+/// Check if the `INK_STATIC_BUFFER_SIZE` is set.
+/// If so, then checks if the current contract has already been compiled with a new value.
+/// If not, or metadata is not present, we need to clean binaries and rebuild.
+fn check_buffer_size_invoke_cargo_clean(
+    crate_metadata: &CrateMetadata,
+    verbosity: &Verbosity,
+) -> Result<()> {
+    if let Ok(buffer_size) = std::env::var("INK_STATIC_BUFFER_SIZE") {
+        let buffer_size_value: u64 = buffer_size
+            .parse()
+            .context("`INK_STATIC_BUFFER_SIZE` must have an integer value.")?;
+
+        let cargo = util::cargo_cmd(
+            "clean",
+            Vec::<&str>::new(),
+            crate_metadata.manifest_path.directory(),
+            *verbosity,
+            vec![],
+        );
+
+        if let Ok(metadata) = ContractMetadata::load(crate_metadata.metadata_path()) {
+            let contract_buffer_size = metadata
+                .abi
+                // get `spec` field
+                .get("spec")
+                .context("spec field should be present in ABI.")?
+                // get `environment` field
+                .get("environment")
+                .context("environment field should be present in ABI.")?
+                // get `staticBufferSize` field
+                .get("staticBufferSize")
+                .context("`staticBufferSize` must be specified.")?
```

Comment:
If the value is not present we would possibly also want to do a clean still?

Reply:
It's always present in the metadata

Reply:
Except when upgrading the version of `ink`, exceptional case I know but the build would fail in that case and require a manual clean of the artifacts? Might as well do it automatically?